lib/internal/Magento/Framework/View/Layout/GeneratorPool.php

```diff
@@ -238,4 +246,17 @@ protected function visibilityConditionsExistsIn(array $data)
             array_key_exists('visibilityConditions', $data['attributes']) &&
             !empty($data['attributes']['visibilityConditions']);
     }
+
+    /**
+     * @param GeneratorInterface $generator
+     * @return bool
+     */
+    private function validateGenerator($generator)
```

Comment:
I think we may simplify this method in the following way:

```
private function validateGenerator($generator)
{
    return $generator instanceof GeneratorInterface;
}
```

Reply:
Hi @dmytro-ch.

I agree with your suggestion and made the modification already.

Thanks.